infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServerCache.java

```diff
@@ -126,28 +73,40 @@ public void delete(@Nonnull ConflictTxStream key, Long globalAddress, @Nonnull R
      * @return global address
      */
     public Long getIfPresent(ConflictTxStream conflictKey) {
-        return conflictCache.getIfPresent(conflictKey);
+        return cacheHash.get(conflictKey);
```

Comment:
please prevent the method returning `null`, use wether Optional or return a default value

Reply:
I don't think Optional should be used in lieu of null "everywhere", especially in a hot code path. The reason being is it adds to memory pressure.

Reply:
Yes, but using `OptionalInt` will be equivalent to using Long. 
If you need better performance then let's return default value and use a primitive long (we need to avoid auto wrapping).
In the other case when we use `Long` is the worst of all possible ways to go.